lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup_facts.py

```diff
@@ -119,20 +135,30 @@
     pass
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi, get_all_objs
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi, get_all_objs, find_dvs_by_name
 
 
 class DVSPortgroupFactsManager(PyVmomi):
     def __init__(self, module):
         super(DVSPortgroupFactsManager, self).__init__(module)
         self.dc_name = self.params['datacenter']
+        self.dvs_name = self.params['dvswitch']
 
     def gather_dvs_portgroup_facts(self):
         datacenter = self.find_datacenter_by_name(self.dc_name)
+
         if datacenter is None:
             self.module.fail_json(msg="Failed to find the datacenter %s" % self.dc_name)
+        if self.dvs_name:
```

Comment:
The logic used to prepare `dvs_lists` increase the global complexity of the `gather_dvs_portgroup_facts()` method. I've the feeling it would be cleaner to create a new new that init `dvs_lists` variable, and call it in the class constructor.

Reply:
what about doing all the logic in the init as implemented in vmware_local_role_facts ?

Already in the code, have a look and let me know